packages/muelu/test/unit_tests/RegionVector.cpp

```diff
@@ -0,0 +1,381 @@
+      TEST_EQUALITY(regVec[0]->getLocalLength(),  9);
+      TEST_EQUALITY(regVec[0]->getGlobalLength(), 36);
+      TEST_EQUALITY(quasiRegVec[0]->getLocalLength(),  9);
+      TEST_EQUALITY(quasiRegVec[0]->getGlobalLength(), 36);
+
+      ArrayRCP<SC> refValues;
+      Teuchos::ArrayRCP<const SC> myValues;
+      if(myRank == 0) {
+        refValues.deepCopy(ArrayView<const SC>({0.0, 1, 2, 3, 4, 5, 6, 7, 8}));
+
+      } else if(myRank == 1) {
+        refValues.deepCopy(ArrayView<const SC>({2.0, 0.1, 1.1, 5.0, 2.1, 3.1, 8.0, 4.1, 5.1}));
+
+      } else if(myRank == 2) {
+        refValues.deepCopy(ArrayView<const SC>({6, 7, 8, 0.2, 1.2, 2.2, 3.2, 4.2, 5.2}));
+
+      } else if(myRank == 3) {
+        refValues.deepCopy(ArrayView<const SC>({8, 4.1, 5.1, 2.2, 0.3, 1.3, 5.2, 2.3, 3.3}));
+      }
+      // Loop over region matrix data and compare it to ref data
+      for (int j = 0; j < maxRegPerProc; j++){
+        myValues = regVec[j]->getData(0);
+        for(int idx = 0; idx < regVec[j]->getLocalLength(); ++idx) {
+          TEST_FLOATING_EQUALITY(myValues[idx], refValues[idx], 100*TMT::eps());
+        }
+        myValues = quasiRegVec[j]->getData(0);
+        for(int idx = 0; idx < quasiRegVec[j]->getLocalLength(); ++idx) {
+          TEST_FLOATING_EQUALITY(myValues[idx], refValues[idx], 100*TMT::eps());
+        }
+      }
+    }
+
+  }
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  {
+    // Create a region vector
+    Array<RCP<Vector> > regVec(maxRegPerProc);
+    for (int j = 0; j < maxRegPerProc; j++) {
+      regVec[j] = VectorFactory::Build(revisedRowMapPerGrp[j]);
+      regVec[j]->putScalar( myRank / 10.0 );
+      for (size_t k = 0; k < regVec[j]->getLocalLength(); ++k){
+        regVec[j]->sumIntoLocalValue(k, k);
+      }
+    }
+
+    // Create a composite vector
+    RCP<Vector> compVec = VectorFactory::Build(dofMap, true);
+
+    regionalToComposite(regVec, compVec, maxRegPerProc, rowMapPerGrp, rowImportPerGrp, Xpetra::ADD);
+
+    if(numRanks == 1) {
+      TEST_EQUALITY(compVec->getLocalLength(),  25);
+      TEST_EQUALITY(compVec->getGlobalLength(), 25);
+
+      ArrayRCP<SC> refValues;
+      Teuchos::ArrayRCP<const SC> myValues;
+      refValues.deepCopy(ArrayView<const SC>({0.0, 1.0, 2.0, 3.0, 4.0,
+                                              5.0, 6.0, 7.0, 8.0, 9.0,
+                                              10.0, 11.0, 12.0, 13.0, 14.0,
+                                              15.0, 16.0, 17.0, 18.0, 19.0,
+                                              20.0, 21.0, 22.0, 23.0, 24.0}));
+      myValues = compVec->getData(0);
+      for(int idx = 0; idx < compVec->getLocalLength(); ++idx) {
+        TEST_FLOATING_EQUALITY(myValues[idx], refValues[idx], 100*TMT::eps());
+      }
+
+    } else if(numRanks == 4) {
+      // All ranks will have the same number of rows/cols/entries
```

Comment:
This comment is misleading. The global length should be the same on all ranks, but the local length differs. Maybe replace it with something along the lines of:
```suggestion
      // All ranks will have the same global length of the vector, but local length differs.
      // Due to region numbering and region interface duplications,
      // lower rank IDs have more local entries then higher rank IDs.
```